From ecashin@coraid.com Tue Jan  3 13:31:21 2006
From: "Ed L. Cashin" <ecashin@coraid.com>
CC: ecashin@coraid.com, Greg K-H <greg@kroah.com>
Subject: aoe: zero packet data after skb allocation
Date: Tue, 03 Jan 2006 16:05:03 -0500
Message-ID: <87hd8l2fb4.fsf@coraid.com>

Zero the data in new socket buffers to prevent leaking information.

Signed-off-by: "Ed L. Cashin" <ecashin@coraid.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/block/aoe/aoecmd.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/block/aoe/aoecmd.c
+++ gregkh-2.6/drivers/block/aoe/aoecmd.c
@@ -28,6 +28,7 @@ new_skb(struct net_device *if_dev, unsig
 		skb->protocol = __constant_htons(ETH_P_AOE);
 		skb->priority = 0;
 		skb_put(skb, len);
+		memset(skb->head, 0, len);
 		skb->next = skb->prev = NULL;
 
 		/* tell the network layer not to perform IP checksums
